lib/ansible/playbook/role/__init__.py

```diff
@@ -125,6 +131,8 @@ def __init__(self, play=None, from_files=None, from_include=False, validate=True
         self._handler_blocks = []
         self._compiled_handler_blocks = None
         self._default_vars = dict()
+        # the _default_vars_full dict includes all default vars, including the full dep chain vars
+        self._default_vars_full: dict[str, str] | None = None  # type: ignore[annotation-unchecked]
```

Comment:
Once #85346 is in, this suppression will no longer be necessary. Additionally, `| None` is implied by the default value of `None` (only works with `None`):
```suggestion
        self._default_vars_full: dict[str, str] = None
```